test/distributed/fsdp/test_fsdp_meta.py

```diff
@@ -59,24 +63,33 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
     def reset_parameters(self, *args, **kwargs):
+        torch.manual_seed(42)
         with torch.no_grad():
-            self.weight.fill_(1)
+            # Use an initialization method that depends on shape
+            torch.nn.init.xavier_uniform_(self.weight, 1.0)
+
+
+class MyBuffer(nn.Module):
+    def __init__(self, device: torch.device):
+        super().__init__()
+        self.register_buffer("buf", torch.empty((3, 3), device=device))
+
+    def reset_parameters(self, *args, **kwargs):
+        torch.manual_seed(42)
+        # Use an initialization method that depends on shape
```

Comment:
curious, what does this change with regard to the test?

Reply:
If we fill with 1 always, then it does not matter if we accidentally re-initialize a nested module that has already been sharded. Here, because the initialization function depends on the parameter shape, the test would fail correctness if we accidentally re-initialized when the parameter had the sharded shape.